### PR TITLE
Add views

### PR DESCRIPTION
**Pull request by bdcostin**

### CODE REVIEW

File: core/views.py [hunk 1]
```diff
@@ -16,17 +18,38 @@ def get_queryset(self):
...
     model = User
     slug_field = 'User.username'
 
 class UserProfileView(generic.FormView, LoginRequiredMixin):
-    model = UserPost
-    template_name = 'user_post'
+    '''Allows a user to view and edit their own profile'''
+    model = User
+    template_name = 'user_profile.html'
     slug_field = 'User.username'
```
Comment (yyapuncich) [inline core/views.py]: Do you have to define slug here- I was planning on adding it to DB with the migration

File: core/views.py [hunk 2]
```diff
@@ -16,17 +18,38 @@ def get_queryset(self):
         """Return the last five published questions."""
         return UserPost.objects.all()
 
+class NewPostView(CreateView, LoginRequiredMixin):
```
Comment (yyapuncich) [inline core/views.py:21]: Is this a view for detail of one post or for the form to create a new post?

File: core/views.py (hunk 1 above)
Comment (bdcostin) [reply core/views.py]: I think it is necessary because the slug is based on the user. What are your thoughts?

File: core/views.py (hunk 2 above)
Comment (bdcostin) [reply core/views.py:21]: It's for the form to create a new post

Comment (yyapuncich) [reply core/views.py:21]: 👍

**Review by yyapuncich — APPROVED** [review]

**Pull request merged**